### PR TITLE
Search implemented and .gitignore added

### PR DESCRIPTION
**Pull request by ALangerbeck**

### CODE REVIEW

File: Search.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        }
+        try {
+            file_scanner = new Scanner(new File(input_file));
+        } catch (FileNotFoundException e) {
+            System.out.println("The file " + input_file + " was not found");
+            return;
+        }
+
+        String currentLine;
+        int line_number = 1;
```
Comment (GustafGG) [inline Search.java:45]: Nitpick: Requirements only stated that you had to print the line, not the line number!
Unnecessary complexity could be argued!

File: Search.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+            input_string = input_string.toLowerCase();
+        }
+        try {
+            file_scanner = new Scanner(new File(input_file));
+        } catch (FileNotFoundException e) {
+            System.out.println("The file " + input_file + " was not found");
+            return;
+        }
+
+        String currentLine;
```
Comment (GustafGG) [inline Search.java:44]: Refactoring string comparison procedure into separate private method would be nice!

File: Search.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+class Search {
+
+    public static void main(String[] args) {
+        boolean case_sensitivity = true;
+        Scanner file_scanner;
+        String input_file = args[1];
+        String input_string = args[0];
```
Comment (GustafGG) [inline Search.java:11]: "input_string" might be too generic name; "input_pattern" would be more clear in my opinion!

File: Search.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+class Search {
+
+    public static void main(String[] args) {
+        boolean case_sensitivity = true;
+        Scanner file_scanner;
+        String input_file = args[1];
```
Comment (GustafGG) [inline Search.java:10]: Input parsing is a little cluttered with variable assignment occuring outside of and within if-statement.
A comment to clarify that this assignment is made with the assumption of case sensitivity being active would perhaps be nice!

I would personally prefer if this assignment was made in the else-statement after the if-statement below, just to further indicate how this input parsing is mutually exclusive from the one in the if-statement!

File: Search.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        String input_string = args[0];
+        if (args.length == 3 && args[0].charAt(0) == '-') {
+            String flag = args[0];
+            input_string = args[1];
+            input_file = args[2];
+
+            switch (flag) {
```
Comment (GustafGG) [inline Search.java:17]: Nice use of switch!

**Review by GustafGG — COMMENTED** [review]
Review done!

File: Search.java (hunk 1 above)
Comment (ALangerbeck) [reply Search.java:45]: Yes will remove

File: Search.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+        } catch (FileNotFoundException e) {
+            System.out.println("The file " + input_file + " was not found");
+            return;
+        }
+
+        String currentLine;
+        int line_number = 1;
+        while (file_scanner.hasNextLine()) {
+            currentLine = file_scanner.nextLine();
+            if (!case_sensitivity && currentLine.toLowerCase().contains(input_string)) {
```
Comment (Filip-Lennhager) [inline Search.java:48]: Duplicate code for the output print, could be in the same if statement.

**Review by GustafGG — APPROVED** [review]
Nicely done -- great fix!

**Pull request merged**